python/pyiceberg/catalog/hive.py

```diff
@@ -201,17 +238,46 @@ def __init__(self, name: str, properties: Properties, uri: str):
         super().__init__(name, properties)
         self._client = _HiveClient(uri)
 
-    def _convert_hive_into_iceberg(self, table: HiveTable) -> Table:
-        # Requires reading the manifest, will implement this in another PR
-        # Check the table type
-        return Table(identifier=(table.dbName, table.tableName))
+    def _convert_hive_into_iceberg(self, table: HiveTable, io: FileIO, metadata_location: Optional[str] = None) -> Table:
+        properties: Dict[str, str] = table.parameters
+        if TABLE_TYPE not in properties:
+            raise NoSuchTableError(f"Property table_type missing, could not determine type: {table.dbName}.{table.tableName}")
+
+        table_type = properties[TABLE_TYPE]
+        if table_type.lower() != ICEBERG:
+            raise NoSuchTableError(f"Property table_type is {table_type}, expected {ICEBERG}: {table.dbName}.{table.tableName}")
+
+        if not metadata_location:
+            if prop_metadata_location := properties.get(METADATA_LOCATION):
+                metadata_location = prop_metadata_location
+            else:
+                raise NoSuchTableError("Metadata location not found")
+
+        file = io.new_input(metadata_location)
+        metadata = FromInputFile.table_metadata(file)
+        return Table(identifier=(table.dbName, table.tableName), metadata=metadata, metadata_location=metadata_location)
+
+    def _write_metadata(self, metadata: TableMetadataV2, io: FileIO, metadata_path: str):
+        ToOutputFile.table_metadata(metadata, io.new_output(metadata_path))
+
+    def _resolve_table_location(self, location: Optional[str], database_name: str, table_name: str):
+        if not location:
+            if warehouse_location := self.properties.get(WAREHOUSE):
+                warehouse_location = warehouse_location.rstrip("/")
+                return f"{warehouse_location}/{database_name}/{table_name}"
+
+            database_properties = self.load_namespace_properties(database_name)
+            if database_location := database_properties.get(LOCATION):
+                database_location = database_location.rstrip("/")
+                return f"{database_location}/{table_name}"
+        raise ValueError("Cannot determine location from warehouse, please provide an explicit location")
```

Comment:
I think indentation for this is wrong. It should be inside the `if not location` part. At this level, we need a `return location`.